base/linalg/diagonal.jl

```diff
@@ -225,6 +225,9 @@ A_mul_B!(A::AbstractMatrix,B::Diagonal)  = scale!(A,B.diag)
 A_mul_Bt!(A::AbstractMatrix,B::Diagonal) = scale!(A,B.diag)
 A_mul_Bc!(A::AbstractMatrix,B::Diagonal) = scale!(A,conj(B.diag))
 
+Ac_mul_B(A::Diagonal,B::AbstractVector) = conj.(A.diag) .* B
```

Comment:
Yes. You can find a couple of issues where the details have been discussed. Please just update this PR instead of opening a new one.